prometheus_client/exposition.py

```diff
@@ -82,10 +83,12 @@ def generate_latest(registry=core.REGISTRY):
 
 
 class MetricsHandler(BaseHTTPRequestHandler):
-    """HTTP handler that gives metrics from ``core.REGISTRY``."""
+    """HTTP handler that gives metrics from ``MetricsHandler.registry``."""
 
     def do_GET(self):
-        registry = core.REGISTRY
+        registry = MetricsHandler.registry
+        if 'prometheus_multiproc_dir' in os.environ:
```

Comment:
This is incorrect, you don't know that the user wants this. They have to add it to the registry themselves.

Reply:
How can I decide whether to run the collector?

Can I add an hook like?

```
if hasattr(registry, 'collect_metrics'):
    registry.collect_metrics(registry)
```

Any hint?

Reply:
How can I decide whether to run this?
```
MultiProcessCollector(registry)
```

Adding a hook?
```
if hasattr(registry, 'collect_metrics'):
    registry.collect_metrics(registry)
```

Hints welcome!

Reply:
In https://github.com/prometheus/client_python#multiprocess-mode-gunicorn the user has to manage the registry themselves, this should work the same way.